components/modal/style.scss

```diff
@@ -0,0 +1,102 @@
+// The scrim behind the modal window.
+.components-modal__screen-overlay {
+	position: fixed;
+	top: 0;
+	right: 0;
+	bottom: 0;
+	left: 0;
+	background-color: rgba( $white, .4 );
+	z-index: z-index( '.components-modal__screen-overlay' );
+
+	// On mobile the main content area has to scroll, otherwise you can invoke the overscroll bounce on the non-scrolling container.
+	@include break-small {
+		position: fixed;
+		top: $admin-bar-height-big;
+	}
+
+	@include break-medium() {
+		top: $admin-bar-height;
+	}
+}
+
+// The modal window element.
+.components-modal__frame {
+	// In small screens the content needs to be full width.
+	position: fixed;
+	top: 0;
+	right: 0;
+	bottom: 0;
+	left: 0;
+	margin: 0;
+
+	// Show slightly bigger on small screens.
+	@include break-small() {
+		position: absolute;
+		right: auto;
+		bottom: auto;
+		max-width: calc( 100% - #{ $panel-padding } - #{ $panel-padding } );
+		margin-right: -50%;
+		transform: translate( -50%, 0 );
+		top: $panel-padding;
+		left: 50%;
+		height: 90%;
+	}
+
+	// Show pretty big on desktop breakpoints.
+	@include break-medium () {
+		max-width: calc( #{ $break-medium } - #{ $panel-padding } - #{ $panel-padding } );
+		transform: translate( -50%, -30% );
+		top: 30%;
+		left: 50%;
+		height: 70%;
+	}
+
+	border: 1px solid $light-gray-500;
+	background-color: $white;
+	box-shadow: $shadow-modal;
+	outline: none;
+}
+
+.components-modal__header {
+	box-sizing: border-box;
+	height: $header-height;
+	border-bottom: 1px solid $light-gray-500;
+	padding: $item-spacing $item-spacing $item-spacing $panel-padding;
+	display: flex;
+	flex-direction: row;
+	align-items: stretch;
+	justify-content: space-between;
+	left: 0;
+	right: 0;
+}
+
+.components-modal__header-heading-container {
+	align-items: center;
+	flex-grow: 1;
+	display: flex;
+	flex-direction: row;
+	justify-content: left;
+}
+
+.components-modal__header-heading {
+	font-size: 1em;
```

Comment:
If the idea here is to assign the text the appearance of "regular text", I wish there were a consolidated way to express this, e.g. via some mixin. There are some variables (e.g. `$default-font-size`) though I'm not sure it's much better than what we have here. Maybe a future improvement (cc @jasmussen).